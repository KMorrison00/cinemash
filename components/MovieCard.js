import { useState } from "react";
import { View, Modal, Text, Image, StyleSheet, TouchableOpacity } from "react-native";
import MovieInfoModal from "./MovieInfoModal";

export default function MovieCard({ movie, height, width }) {
    // image source URL
    const imageSrc = `https://image.tmdb.org/t/p/w500${movie.backdrop_path}`;
    
    // Define a state variable to control the visibility of the modal
    const [visible, setVisible] = useState(false);

    const openModal = () => setVisible(true);
    const closeModal = () => setVisible(false);

    return (
        <>
            <Modal
                visible={visible}
                transparent={true}
                onRequestClose={closeModal}
                animationType="slide"
            >
                {/* Touchable area to close the modal */}
                <TouchableOpacity onPress={closeModal} style={{ height: "100%" }} />
                
                {/* Custom MovieInfoModal component */}
                <MovieInfoModal movie={movie} />
            </Modal>

            {/* Touchable area to open the modal */}
            <TouchableOpacity onPress={openModal}>
                <View style={styles.movieCard}>
                    {/* If movie has a backdrop_path, display the image, otherwise display the movie title */}
                    {movie.backdrop_path
                        ? <Image
                            style={{ height, width, ...styles.image }}
                            source={{ uri: imageSrc }}
                        />
                        : <View style={{ height, width, ...styles.viewAlt }}>
                            <Text style={styles.textAlt}>{movie.title}</Text>
                        </View>
                    }
                </View>
            </TouchableOpacity>
        </>
    );
}
const styles = StyleSheet.create({
    movieCard: {
        marginVertical: 14,
    },
    title: {
        color: "white"
    },
    image: {
        borderRadius: 10,
        borderWidth: 2,
        borderColor: "#252525",
    },
    textAlt: {
        color: "#FFF",
        textAlign: "center",
        fontSize: 25,
        fontWeight: "400",
    },
    viewAlt: {
        padding: 2,
        borderColor: "#FFF",
        borderWidth: 4,
        borderRadius: 10,
        display: "flex",
        justifyContent: "center",
    }
})

