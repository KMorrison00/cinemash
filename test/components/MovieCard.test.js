import React from 'react';
import { TouchableOpacity, Modal } from 'react-native';
import MovieCard from '../../components/MovieCard';
import renderer from 'react-test-renderer';

describe('MovieCard', () => {
    it('renders correctly', () => {
        const movie = {
            backdrop_path: '/4MCKNAc6AbWjEsM2h9Xc29owo4z.jpg',
            title: 'The Beekeeper',
        };

        const tree = renderer
            .create(<MovieCard movie={movie} height={100} width={100} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('opens and closes modal', () => {
        const movie = {
            backdrop_path: '/4MCKNAc6AbWjEsM2h9Xc29owo4z.jpg',
            title: 'The Beekeeper',
        };

        const testRenderer = renderer.create(<MovieCard movie={movie} height={100} width={100} />);
        const testInstance = testRenderer.root;

        // Find the TouchableOpacity components
        const touchableOpacities = testInstance.findAllByType(TouchableOpacity);

        // Check if the TouchableOpacity components exist
        if (touchableOpacities.length > 1) {
            // Simulate a press to open the modal
            touchableOpacities[1].props.onPress();

            // Check if modal is visible
            let modal = testInstance.findByType(Modal);
            expect(modal.props.visible).toBe(true);

            // Simulate a press to close the modal
            touchableOpacities[1].props.onPress();

            // Check if modal is not visible
            modal = testInstance.findByType(Modal);
            expect(modal.props.visible).toBe(false);
        }
    });
})