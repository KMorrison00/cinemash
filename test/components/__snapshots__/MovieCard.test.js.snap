// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "marginVertical": 14,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://image.tmdb.org/t/p/w500/4MCKNAc6AbWjEsM2h9Xc29owo4z.jpg",
        }
      }
      style={
        {
          "borderColor": "#252525",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
